File: python/001.py
from itertools import ifilter

def solve(N, d=[3, 5]):
    def f(x):
        return any(ifilter(lambda dd: x and x % dd == 0, d))

    return sum([i for i in range(N) if f(i)])

if __name__ == '__main__':
    assert solve(10) == 23

    assert solve(1000) == 233168

File: ruby/001.rb

def solve(n, divs)
    (0...n).reduce do |sum, i|
        if divs.any?{|x| i % x == 0}
           sum += i
        end
        sum
    end
end


require 'test/unit'
class TestSolve < Test::Unit::TestCase
    def test_10
        assert solve(10, [3, 5]) == 23
    end

    def test_1000
        assert solve(1000, [3, 5]) == 233168
    end
end
